libraries/extractor/src/main/java/androidx/media3/extractor/mp4/AtomParsers.java

```diff
@@ -1122,6 +1123,133 @@ private static void parseTextSampleEntry(
             .build();
   }
 
+  // Helper class for parsing syntax elements from AV1 bitstream.
+  // Bitstream specification: https://aomediacodec.github.io/av1-spec/av1-spec.pdf
+  private static final class Av1BitstreamParser {
+    private static final String TAG = "Av1BitstreamParser";
+
+    private ParsableByteArray data;
+    private int bitCount;
+    private int currentByte;
+
+    public int colorDescriptionPresentFlag;
+    public int colorPrimaries;
+    public int transferCharacteristics;
+    public int matrixCoefficients;
+    public int colorRange;
+
+    public Av1BitstreamParser(ParsableByteArray data) {
+      this.data = data;
+    }
+
+    private int f(int length) {
+      // 4.10.2. f(n): Unsigned n-bit number appearing directly in the bitstream.
+      // The bits are read from high to low order.
+      int result = 0;
+      while (length > 0) {
+        if (bitCount == 0) {
+          currentByte = data.readUnsignedByte();
+          bitCount = 8;
+        }
+        int newBitCount = min(length, bitCount);
+        bitCount -= newBitCount;
+        length -= newBitCount;
+        result = (result << newBitCount) |
+            ((currentByte >> bitCount) & ((1 << newBitCount) - 1));
+      }
+      return result;
+    }
+
+    public boolean parseSequenceHeader() {
```

Comment:
Javadoc describing method and its usage. It also seems that this parseSequenceHeader can only be called once? Can you add a private boolean parseSequenceHeaderCalled field? Then in the first line in this method call "CHECK_STATE(!parseSequenceHeaderCalled)" with the subsequent line setting parseSequenceHeaderCalled to true?

Reply:
updated Javadoc and made sure it can only be called once. however, I didn't find where CHECK_STATE is defined so I just let the method return false instead.